ext/canvas/lib.rs

```diff
@@ -118,24 +121,21 @@ struct DecodedPng {
 #[op2]
 #[serde]
 fn op_image_decode_png(#[buffer] buf: &[u8]) -> Result<DecodedPng, AnyError> {
-  let png = image::codecs::png::PngDecoder::new(buf)?;
-
-  let (width, height) = png.dimensions();
+  let reader = BufReader::new(buf);
+  let decoder = PngDecoder::new(reader)?;
+  let image = DynamicImage::from_decoder(decoder)?;
+  let (width, height) = image.dimensions();
 
   // TODO(@crowlKats): maybe use DynamicImage https://docs.rs/image/0.24.7/image/enum.DynamicImage.html ?
-  if png.color_type() != ColorType::Rgba8 {
+  if image.color() != ColorType::Rgba8 {
     return Err(type_error(format!(
       "Color type '{:?}' not supported",
-      png.color_type()
+      image.color()
     )));
   }
 
-  let mut png_data = Vec::with_capacity(png.total_bytes() as usize);
-
-  png.read_image(&mut png_data)?;
```

Comment:
Ah, I got the detail of the stack trace. https://github.com/denoland/deno/pull/23518#issuecomment-2073974039

The `png_data.len()` returns 0 by `Vec::with_capacity`.
https://doc.rust-lang.org/std/vec/struct.Vec.html#examples-2

And the `read_image` assert it inside.
https://github.com/image-rs/image/blob/2b513ae9a6ac306e752914f562e7d408f096ba3f/src/codecs/png.rs#L268-L271

So, it will pass with adding the filling 0 of vector like below.
```diff rust
  let mut png_data = Vec::with_capacity(png.total_bytes() as usize);
+ png_data.resize(png.total_bytes() as usize, 0);

  png.read_image(&mut png_data)?;
...
```

Reply:
fix it